### DISCUSSION TITLE
Groovy code to add image to a node

### DISCUSSION THREAD

**Discussion by biswajit-goswami** [freeplane/freeplane#1622] [Help] [answered]
I have an image file , eg C:/myimages/image.png and I need to add this image to a selected node ( not link, but view the image in the node) . I am using groovy script and wondering what property to set in the c.selected node.
if I do c.selected.link.text and assign the file path via file://c:/...... , it creates a link..but I want to view the image ..same as achieved by the Add image option in the menu bar.
any inputs?

**Comment by EdoFro** [comment]
use [node.externalObject](https://docs.freeplane.org/api/org/freeplane/api/NodeRO.html#getExternalObject())

and one of its [setter methods](https://docs.freeplane.org/api/org/freeplane/api/ExternalObject.html)

example:
```groovy
def myUri ='file://c:/myimages/image.png'

node.externalObject.uri = new URI(myUri)
```

**Reply by biswajit-goswami** [reply, marked as answer]
thanks for the hint. The above did not work as it gave unknown host exception ..checked the docs for externalObject and then the below worked:
def myFile="c:/myimages/image.png"
node.externalObject.file=new File(myFile)
